lib/igniter.ex

```diff
@@ -330,32 +340,56 @@ defmodule Igniter do
   end
 
   @doc """
-  Finds the `Igniter.Mix.Task` task by name and composes it (calls its `igniter/2`) into the current igniter.
-  If the task doesn't exist, a fallback implementation may be provided as the last argument.
+  Finds the `Igniter.Mix.Task` task by name and composes it with `igniter`.
+
+  This will call the task's `igniter/1` (or `igniter/2`) callback, setting
+  `igniter.args` using the current `igniter.args.argv_flags`. This prevents
+  composed tasks from accidentally consuming positional arguments. If you
+  wish the composed task to access positional arguments, you must call this
+  function with `igniter.args.argv` as the `argv` argument.
+
+  If the task doesn't exist, a `fallback` function may be provided. This
+  function should accept and return the `igniter`.
```

Comment:
LGTM! It is a common pattern to decide add your own positional arguments to a composed task, so perhaps we should show that as an example:

```elixir
Igniter.compose_task("task", ["positional", "argument"] ++ igniter.args.argv_glags)
```

We use this in the Ash installer when the `--example` flag is given, we compose the `ash.gen.resource` command twice.